src/HTTP/Factory.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/* Copyright (c) 2016 Fabian Schmid <fs@studer-raimann.ch> Extended GPL, see docs/LICENSE */
+namespace ILIAS\HTTP;
+
+use ILIAS\HTTP\Cookies\CookieJar;
+use ILIAS\HTTP\Response\rendering\RenderingException;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * Interface Factory
```

Comment:
Maybe a tremendously stupid question, but while browsing this I did not really get that: Where and how will this factory be implemented? By whom?